include/uv.h

```diff
@@ -579,6 +580,28 @@ UV_EXTERN int uv_tcp_connect(uv_connect_t* req,
                              const struct sockaddr* addr,
                              uv_connect_cb cb);
 
+/*
+ * uv_stream_info_t is used to store exported stream (using uv_export),
+ * which can be imported into a different event-loop within the same process
+ * (using uv_import).
+ */
+struct uv_stream_info_s {
+  uv_handle_type type;
+  UV_STREAM_INFO_PRIVATE_FIELDS
+};
+
+/*
+ * Exports uv_stream_t as uv_stream_info_t value, which could
+ * be used to initialize shared streams within the same process.
+ */
+UV_EXTERN int uv_export(uv_stream_t* stream, uv_stream_info_t* info);
```

Comment:
I think this needs to operate on `uv_handle_t` (and `uv_stream_info_t` be renamed accordingly) because e.g. a `uv_udp_t` handle is not an instance of `uv_stream_t`.

For handles where it's not feasible or sensible to export them (like timers), it's okay to simply return `UV_EINVAL`.

Some thought should be given on how to extend the API later on. For example, there should be room in the struct to store the relevant data for UDP and pipe handles.